reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -237,6 +237,52 @@ bool ReconfigurationHandler::handle(const concord::messages::CreateDbCheckpointC
   }
 }
 
+bool ReconfigurationHandler::handle(const concord::messages::StateSnapshotRequest& cmd,
+                                    uint64_t sequence_number,
+                                    uint32_t,
+                                    const std::optional<bftEngine::Timestamp>& timestamp,
+                                    concord::messages::ReconfigurationResponse& rres) {
+  if (!bftEngine::ReplicaConfig::instance().dbCheckpointFeatureEnabled) {
+    const auto err = "StateSnapshotRequest(participant ID = " + cmd.participant_id +
+                     "): failed, the DB checkpoint feature is disabled";
+    LOG_WARN(getLogger(), err);
+    rres.response = concord::messages::ReconfigurationErrorMsg{err};
+    return false;
+  }
+
+  auto resp = StateSnapshotResponse{};
+  const auto last_checkpoint_desc = DbCheckpointManager::instance().getLastCreatedDbCheckpointMetadata();
+  // TODO: We currently only support new participants and, therefore, the event group ID will always be the one before
+  // the first one, namely 0.
+  if (last_checkpoint_desc) {
+    resp.data.emplace();
+    resp.data->snapshot_id = last_checkpoint_desc->checkPointId_;
+    resp.data->event_group_id = 0;
+    LOG_INFO(getLogger(),
+             "StateSnapshotRequest(participant ID = " << cmd.participant_id << "): using existing last checkpoint ID: "
+                                                      << last_checkpoint_desc->checkPointId_);
+  } else {
+    const auto checkpoint_id = DbCheckpointManager::instance().createDbCheckpointAsync(sequence_number, timestamp);
```

Comment:
note: DbCheckpointManager does not create db checkpoint if `ReplicaConfig::maxNumberOfDbCheckpoints == 0`
We should check this at the start along with feature flag and return error in response

Reply:
I see, will do so, thanks. I am wondering if we need both the flag and the number of checkpoints in that case, though. Do we have a use case for setting the max number to a non-zero value, yet having the flag set to false?

Reply:
feature_flag = True && num_of_db_checkpoints = 0; //possible, for example we create db checkpoints on (f+1) replicas only. But all or at-least 2f+1 replicas participate in consensus to decide the seq num to initiate creation of db checkpoint at and that will be based on the flag

Reply:
Got it, thanks!